src/main/CommandHandler.cpp

```diff
@@ -743,95 +744,136 @@ CommandHandler::sorobanInfo(std::string const&, std::string& retStr)
+            res["fee_historical_1kb"] =
+                static_cast<Json::Int64>(conf.feeHistorical1KB());
+
+            // Contract events settings
+            res["tx"]["max_contract_events_size_bytes"] =
+                conf.txMaxContractEventsSizeBytes();
+            res["fee_contract_events_size_1kb"] =
+                static_cast<Json::Int64>(conf.feeContractEventsSize1KB());
+
+            // Bandwidth related data settings
+            res["ledger"]["max_tx_size_bytes"] =
+                conf.ledgerMaxTransactionSizesBytes();
+            res["tx"]["max_size_bytes"] = conf.txMaxSizeBytes();
+            res["fee_transaction_size_1kb"] =
+                static_cast<Json::Int64>(conf.feeTransactionSize1KB());
+
+            // General execution ledger settings
+            res["ledger"]["max_tx_count"] = conf.ledgerMaxTxCount();
+
+            // State archival settings
+            auto& archivalInfo = res["state_archival"];
+            auto const& stateArchivalSettings = conf.stateArchivalSettings();
+            archivalInfo["max_entry_ttl"] = stateArchivalSettings.maxEntryTTL;
+            archivalInfo["min_temporary_ttl"] =
+                stateArchivalSettings.minTemporaryTTL;
+            archivalInfo["min_persistent_ttl"] =
+                stateArchivalSettings.minPersistentTTL;
+
+            archivalInfo["persistent_rent_rate_denominator"] =
+                static_cast<Json::Int64>(
+                    stateArchivalSettings.persistentRentRateDenominator);
+            archivalInfo["temp_rent_rate_denominator"] =
+                static_cast<Json::Int64>(
+                    stateArchivalSettings.tempRentRateDenominator);
+
+            archivalInfo["max_entries_to_archive"] =
+                stateArchivalSettings.maxEntriesToArchive;
+            archivalInfo["bucketlist_size_window_sample_size"] =
+                stateArchivalSettings.bucketListSizeWindowSampleSize;
+
+            archivalInfo["eviction_scan_size"] = static_cast<Json::UInt64>(
+                stateArchivalSettings.evictionScanSize);
+            archivalInfo["starting_eviction_scan_level"] =
+                stateArchivalSettings.startingEvictionScanLevel;
+
+            // non-configurable settings
+            archivalInfo["bucket_list_size_snapshot_period"] =
+                conf.getBucketListSizeSnapshotPeriod();
+            archivalInfo["average_bucket_list_size"] =
+                static_cast<Json::UInt64>(conf.getAverageBucketListSize());
+            retStr = res.toStyledString();
+        }
+        else if (format == "detailed")
+        {
+            LedgerTxn ltx(mApp.getLedgerTxnRoot(),
+                          /* shouldUpdateLastModified */ false,
+                          TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
+            xdr::xvector<ConfigSettingEntry> entries;
+            for (uint32_t i = 0;
+                 i <= static_cast<uint32_t>(CONFIG_SETTING_EVICTION_ITERATOR);
```

Comment:
Nit: `for (auto t : xdr::xdr_traits<ConfigSettingID>::enum_values())`

Reply:
fixed